### PR TITLE
vagrant: 2.2.3 -> 2.2.5

### LINKED ISSUE #64243: Vagrant is out-of-date (2.2.3 -> 2.2.5)
I'm creating this issue in case anyone wants to take a look.

In the expression there is written:

```
  # NOTE: bumping the version and updating the hash is insufficient;
  # you must use bundix to generate a new gemset.nix in the Vagrant source.
```

I've already used bundix to update a gem, but I don't see a Gemfile in the vagrant folder (only a gemset.nix), so I'm lost 😔

### PR DESCRIPTION
**Pull request by angristan**
###### Motivation for this change

Fixes #64243.

FYI, it is written:

>  NOTE: bumping the version and updating the hash is insufficient;
> you must use bundix to generate a new gemset.nix in the Vagrant source.

However doing this did work for me:

```
stanislas@nixpsla ~/n/p/d/t/vagrant> (vagrant/update/2.2.5|✚1) ./result/bin/vagrant version
Installed Version: 2.2.5
Latest Version: 2.2.5
 
You're running an up-to-date version of Vagrant!
```

Now, I am interested to know how to update the gems without the Gemfile. I saw https://github.com/NixOS/nixpkgs/pull/47519#issuecomment-425645412 but it's doing weird stuff for me.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @aneeshusa

### CODE REVIEW

File: pkgs/development/tools/vagrant/default.nix [hunk 1]
```diff
@@ -4,9 +4,9 @@
 let
   # NOTE: bumping the version and updating the hash is insufficient;
   # you must use bundix to generate a new gemset.nix in the Vagrant source.
```
Comment (etu) [inline pkgs/development/tools/vagrant/default.nix]: I feel that we're missing changes needed in this PR.

**Review by etu — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/vagrant/default.nix (hunk 1 above)
Comment (angristan) [reply pkgs/development/tools/vagrant/default.nix]: Yes, see my original comment (I was editing it). I would like to get help on this! :)

**Comment by angristan** [comment]
FYI by using Vagrant's Gemfile + bundle + bundix, I'm getting this: [gemset.nix.txt](https://github.com/NixOS/nixpkgs/files/3359572/gemset.nix.txt), which seems weirdly... long?

**Comment by worldofpeace** [comment]
Perhaps @alyssais and @manveru could help you with this.

I don't think I see any reason that the `gemset.nix` would be generated improperly.

**Comment by angristan** [comment]
In any case, we need some clear documentation/script to update vagrant once this is settled! :slightly_smiling_face:

**Comment by manveru** [comment]
Well, I can explain it a bit later, a working version is at: https://github.com/manveru/nixpkgs/commit/c2b5d07e6126202c342e7d8b53e5b283a9064d52 for now
You basically have to make a `Gemfile` because bundix doesn't support the `gemspec` method yet. It looks like this:

```
# frozen_string_literal: true

source 'https://rubygems.org' do
  gem 'bcrypt_pbkdf', '~> 1.0.0'
  gem 'childprocess', '~> 0.6.0'
  gem 'ed25519', '~> 1.2.4'
  gem 'erubis', '~> 2.7.0'
  gem 'hashicorp-checkpoint', '~> 0.1.5'
  gem 'i18n', '~> 1.1.1'
  gem 'listen', '~> 3.1.5'
  gem 'log4r', '~> 1.1.9', '< 1.1.11'
  gem 'net-scp', '~> 1.2.0'
  gem 'net-sftp', '~> 2.1'
  gem 'net-ssh', '~> 5.1.0'
  gem 'rb-kqueue', '~> 0.2.0'
  gem 'rest-client', '>= 1.6.0', '< 3.0'
  gem 'ruby_dep', '<= 1.3.1'
  gem 'rubyzip', '~> 1.2.2'
  gem 'vagrant-spec', git: 'https://github.com/hashicorp/vagrant-spec.git'
  gem 'vagrant_cloud', '~> 2.0.2'
  gem 'wdm', '~> 0.1.0'
  gem 'winrm', '~> 2.1'
  gem 'winrm-elevated', '~> 1.1'
  gem 'winrm-fs', '~> 1.0'
end
```

is taken from the dependencies in the Gemfile and gemspec files in the vagrant repo.
Then you can run `nix-shell --pure -p bundler bundix git --run 'bundle lock && bundix'` to generate the `gemspec.nix`.

**Comment by angristan** [comment]
Thanks for posting this! :slightly_smiling_face: 

If I understand correctly you manually created a `Gemfile` using:

https://github.com/hashicorp/vagrant/blob/5d4c9ef8a74df967f6f24867360d4f6e1f3705ff/vagrant.gemspec#L19..#L43

Thus... don't we keep our own Gemfile (or why did we remove it) in `nixpkgs/pkgs/development/tools/vagrant/Gemfile`? It seems very bothersome to manually create a new one each time (+ how are newcomers supposed to know?)

Then, I suppose there are some missing steps:

```
stanislas@nixpsla ~/n/p/d/t/vagrant> (vagrant/update/2.2.5|…) nix-shell --pure -p bundler bundix git --run 'bundle lock && bundix'
Fetching https://github.com/hashicorp/vagrant-spec.git
There was an error accessing `/nix/store/glhmssxi9zpw6r84ws2zdyw2hfvpsnm2-ruby-2.5.5/lib/ruby/gems/2.5.0/cache/bundler/git`.
The underlying system error is Errno::EROFS: Read-only file system @ dir_s_mkdir -
/nix/store/glhmssxi9zpw6r84ws2zdyw2hfvpsnm2-ruby-2.5.5/lib/ruby/gems/2.5.0/cache/bundler
```
When running `bundle install` the "hack" I found was to specify a custom `--path` dir. How did you do?

**Comment by Ma27** [comment]
@GrahamcOfBorg build vagrant

**Review by Ma27 — APPROVED** [review]
Built and tested locally. Regarding the `Gemfile` discussion: I agree that we should either add a `Gemfile` for vagrant and/or leave a comment on how to regenerate vagrant's `gemset.nix`.

However I'd prefer to have an up-to-date version of vagrant on master and release-19.03 now, so I'll merge this for now, feel free to improve the documentation for this package in a separate PR and ping me :)

**Comment by Ma27** [comment]
@angristan thanks!
Backported as 25eb016a2f51c0afd6b84c45f4d1ee96ef5aeeba

**Comment by angristan** [comment]
Ok, thanks!

**Comment by angristan** [comment]
I think I will do a PR to add the Gemfile back. What would be a good place to add documentation (besides a comment in the Gemfile?)

**Comment by Ma27** [comment]
> I think I will do a PR to add the Gemfile back. What would be a good place to add documentation (besides a comment in the Gemfile?)

As this is only relevant for folks who want to modify the vagrant package it should be sufficient to enhance the comment :)

**Comment by angristan** [comment]
After thinking a bit about it, adding an update script is better than revendoring `Gemfile` and `Gemfile.lock`. See #65800 (need some feedback!)

**Pull request merged**